### PR TITLE
[#3809] Fix change view of submission

### LINKED ISSUE #3809: Unhandled exception when accessing an non existing submission detail page through the admin
Sentry 341140

https://github.com/open-formulieren/open-forms/blob/c12c51b94c55895490ba8e8c671a060d080e1586/src/openforms/submissions/admin.py#L269-L275

`get_object` returns an instance of the model or `None`. Probably just a matter of raising a 404.

(This can happen when accessing the (deleted) submission from the submission report detail page.

### PR DESCRIPTION
**Pull request by Viicos**
Fixes #3809

### CODE REVIEW

File: src/openforms/submissions/tests/test_admin.py [hunk 1]
```diff
@@ -293,6 +293,13 @@ def test_search(self):
 
         self.assertEqual(200, response.status_code)
 
+    def test_change_view(self):
+        breakpoint()
```
Comment (vaszig) [inline src/openforms/submissions/tests/test_admin.py:297]: Let's remove this

**Review by vaszig — APPROVED** [review]
As discussed in the office as well, the report is deleted along with the submission deletion so it's probably an edge case, but anyway this fix makes sense generally.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]
